pkg/version/compare.go

```diff
@@ -131,15 +141,41 @@ func CompareVersionsOr(versionA, versionB string, incomparableRes int) int {
 	return incomparableRes
 }
```

Comment:
I have doubts this function did the right thing before the PR, I think, it may do even more wrong thing after. The presence of tests doesn't make things more clear because there's little semantics behind the test cases and their excessive amount can hide entire groups untested.

My preliminary suggestion would be to do a pre-factor to make `compare.go` and `compare_test.go` more clear and more well tested before applying changes of this PR.
1. Drop tests (and possibly code) that supports versions that consist of four groups `2.5.n.m` and `3.0.n.m`. The last one was `3.0.62.1` around August 2021. I don't think there's anyone who still uses and pays for that.
2. Cover `getEffectVersion` (should it be called `getEffectiveVersion`?) with its own independent tests giving it each type of version.
3. Change `TestCompareReleaseVersion` to call `CompareReleaseVersionOr` with no-zero `incomparableRes` and assert when we receive true-zero versus our `incomparableRes`.
4. Change `getEffectVersion` to return int slice instead of string and `if kindA == kindB` clause in this `CompareVersionsOr` to call `lexicographicCompareIntArrays` instead of `CompareReleaseVersionOr`.
5. Split up `CompareVersionsOr` tests into groups according to what cases they check.